src/changelog/prepareChangelogPaths.ts

```diff
@@ -41,30 +39,26 @@ export function prepareChangelogPaths(params: {
 
     const defaultPath = path.join(cwd, `CHANGELOG.${ext}`);
 
-    if (uniqueFilenames) {
-      // Filenames should use a suffix. Start by getting any existing suffixed changelogs...
-      existingSuffixedPaths ??= _getExistingSuffixedChangelogs(cwd);
-      // Use the existing suffix or generate a new one
-      paths.suffix ??= existingSuffixedPaths.suffix ?? uuid().slice(0, 8);
-
-      if (existingSuffixedPaths[ext]) {
-        // Suffixed file already exists, so use it
-        paths[ext] = existingSuffixedPaths[ext];
-      } else {
-        // New suffixed file
-        paths[ext] = path.join(cwd, `CHANGELOG-${paths.suffix}.${ext}`);
-        // Move any existing non-suffixed file to the new suffixed path
-        moveIfNeeded({ oldPath: defaultPath, newPath: paths[ext]!, hadSuffix: false });
-      }
-    } else {
-      // Filenames should not use a suffix
-      paths[ext] = defaultPath;
-
-      if (!fs.existsSync(defaultPath)) {
-        // If the default file doesn't exist, check for a suffixed file to move back
-        existingSuffixedPaths ??= _getExistingSuffixedChangelogs(cwd);
-        if (existingSuffixedPaths[ext]) {
-          moveIfNeeded({ oldPath: existingSuffixedPaths[ext]!, newPath: defaultPath, hadSuffix: true });
+    // Generate a unique filename based on the package name hash.
+    const hash = crypto.createHash('md5').update(packageName).digest('hex').slice(0, 8);
```

Comment:
Thank you for this change. This will avoid the possibility of some corner cases with the random method. There may even be some benefits if a package is moved within the repo (without renaming the package).